src/v/serde/version.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cinttypes>
+
+namespace serde {
+
+using version_t = std::uint8_t;
+
+template<version_t V>
+struct version {
```

Comment:
I wonder if we really need this in a separate header. These types are really part of the envelope definition.